test/tstInterpMovingLeastSquares.cpp

```diff
@@ -211,34 +213,63 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(moving_least_square_cartesian_convergence,
         target_coords(2) = {.788675, .211325};
         target_coords(3) = {.211325, .211325};
         for (int i = 0; i < num_targets; ++i)
-          target_values(i) = f(target_coords(i)[0], target_coords(i)[1]);
+          target_values(i) = f(target_coords(i));
       });
+  auto host_target_coords = Kokkos::create_mirror_view(target_coords);
 
-  constexpr int num_refinements = 9;
-  constexpr int n = (1 << num_refinements) + 1;
-  constexpr double h = 2. / (n - 1);
-  constexpr int num_sources = n * n;
+  std::vector<double> expected_errors({
+      2.e-7,
+      5.e-8,
+      1.e-8,
+      7.e-10,
+      2.e-10,
+      1.e-11,
+      3.e-12,
+      3.e-13,
+      4.e-14,
+      5.e-15,
+  });
 
-  std::cout << "h = " << h << std::endl;
+  for (int num_refinements = 9; num_refinements < 19; ++num_refinements)
+  {
+    int n = (1 << num_refinements) + 1;
+    double h = 2. / (n - 1);
 
-  Kokkos::View<Point *, MemorySpace> source_coords("source_coords",
-                                                   num_sources);
-  Kokkos::View<double *, MemorySpace> source_values("source_values",
-                                                    num_sources);
-  Kokkos::parallel_for(
-      Kokkos::MDRangePolicy(space, {0, 0}, {n, n}),
-      KOKKOS_LAMBDA(int const i, int const j) {
-        int index = i * n + j;
-        source_coords(index) = {-1 + i * h, -1 + j * h};
-        source_values(index) =
-            f(source_coords(index)[0], source_coords(index)[1]);
-      });
+    // We construct 25 points around each target point to avoid constructing a
+    // full mesh grid. 25 points is sufficient to find the 6 nearest neighbors
+    // needed for the 2nd-order polynomial basis.
+    int const n_points_per_target_1d = 5;
+    int const n_points_per_target =
+        n_points_per_target_1d * n_points_per_target_1d;
+    Kokkos::View<Point *, MemorySpace> source_coords(
+        "source_coords", n_points_per_target * num_targets);
+    Kokkos::View<double *, MemorySpace> source_values(
+        "source_values", n_points_per_target * num_targets);
+    Kokkos::parallel_for(
+        Kokkos::RangePolicy(space, 0, num_targets),
+        KOKKOS_LAMBDA(int target_index) {
+          int const start_x = host_target_coords[target_index][0] / h;
+          int const start_y = host_target_coords[target_index][1] / h;
+          for (int i = -2; i <= 2; ++i)
+            for (int j = -2; j <= 2; ++j)
+            {
+              int const linear_index = target_index * n_points_per_target +
+                                       (i + 2) * n_points_per_target_1d +
+                                       (j + 2);
+              source_coords(linear_index) = {(start_x + i) * h,
+                                             (start_y + j) * h};
+              source_values(linear_index) = f(source_coords(linear_index));
+            }
+        });
```

Comment:
Wonder if something like this would be easier to digest. Not sure. 
```suggestion
    int const num_sources_per_target = 25;
    int const num_sources = num_targets * num_sources_per_target;
    Kokkos::View<Point *, MemorySpace> source_coords(
        "source_coords", num_sources);
    Kokkos::View<double *, MemorySpace> source_values(
        "source_values", num_sources);
    Kokkos::parallel_for(
        Kokkos::RangePolicy(space, 0, num_sources),
        KOKKOS_LAMBDA(int source_index) {
          auto const target_index = source_index / num_sources_per_target;
          auto const k = Kokkos::sqrt(num_sources_per_target);
          auto const i = source_index % k;
          auto const j = (source_index / k) % k;
          
          int const start_x = target_coords(target_index)[0] / h;
          int const start_y = target_coords(target_index)[1] / h;
          source_coords(source_index) = {(start_x + i - k/2) * h, (start_y + j - k/2) * h};
          source_values(souce_index) = f(source_coords(source_index));
        });
```